ConfigSpace/configuration_space.pyx

```diff
@@ -1341,6 +1344,19 @@ class ConfigurationSpace(collections.abc.Mapping):
         """
         self.random = np.random.RandomState(seed)
 
+    def to_uniform(self) -> 'ConfigurationSpace':
+        uniform_config_space = ConfigurationSpace()
+        for parameter in self.get_hyperparameters():
+            if type(parameter) not in [UniformFloatHyperparameter, UniformIntegerHyperparameter, OrdinalHyperparameter]:
+                uniform_config_space.add_hyperparameter(parameter.to_uniform())
+            else:
+                uniform_config_space.add_hyperparameter(parameter)
+
+        # TODO - check that this is certainly correct    
+        uniform_config_space.add_conditions(self.get_conditions())
+        uniform_config_space.add_forbidden_clauses(self.get_forbiddens())
+        
+        return uniform_config_space
```

Comment:
I have to check this later, you add the uniform version of parameters but do you require the non_uniform parameters to be removed from the config space?

Reply:
Sure, this is a bit unintuitive. Let's see if I understood you correctly:

I'm generating a ConfigurationSpace of only uniform parameters (or categoricals without weights in that specific case). However, the original one is also kept, and both are fed into SMAC. The reason for this is that the local search associated with PiBO samples a fraction of its starting points uniformly, and the other half from the prior (which has proven empirically superior to only doing one or the other). As such, we need two ConfigurationSpaces.

Let me know if I didn't answer your question!

Reply:
Sorry, I missed the very first line of the function call. This seems correct except for your TODO which is something to test.

I think `config_space.to_uniform()` is not very intuitive of it's functionality.
What does it mean for a `config_space` to be _uniform_? While this interpretation makes sense, there could be other interpretations that make sense.

Considering it's one of the main API points for external uses, it could do with some documentation on that point.

If it's also a major API point, it could do with a little example with explanation in [the docs](https://github.com/hvarfner/ConfigSpace/blob/betaparams/docs/source/User-Guide.rst) (which are sorely in need of some updating).

Reply:
Ah, that makes sense. Yeah, I think a more precise name is a good idea. I could definitely do some docs writing, as well.

I think something like remove_parameter_priors or make_parameters_uniform is ultimately a more descriptive name for the method.

Reply:
That's true. Will fix!